core/variant/variant.h

```diff
@@ -807,7 +807,21 @@ const Variant::ObjData &Variant::_get_obj() const {
 	return *reinterpret_cast<const ObjData *>(&_data._mem[0]);
 }
 
-String vformat(const String &p_text, const Variant &p1 = Variant(), const Variant &p2 = Variant(), const Variant &p3 = Variant(), const Variant &p4 = Variant(), const Variant &p5 = Variant());
+template <typename... VarArgs>
+String vformat(const String &p_text, const VarArgs... p_args) {
+	Variant args[sizeof...(p_args) + 1] = { p_args..., Variant() }; // +1 makes sure zero sized arrays are also supported.
+	Array args_array;
+	for (uint32_t i = 0; i < sizeof...(p_args); i++) {
+		args_array.push_back(args[i]);
+	}
```

Comment:
Instead of pushbacking and resizing array for each element, you could resize it beforehand and do `args_array[i] = args[i]`. That's what all other vararg functions do.

Reply:
How is this now?

Reply:
The new +1 is sus, I added a comment below.